File: top150/valid_anagram.go
func isAnagram(s string, t string) bool {
	if len(s) != len(t) {
		return false
	}

	sMap := make(map[byte]int)
	for i := range s {
		if s[i] != t[i] {
			sMap[s[i]]++
			sMap[t[i]]--
		}
	}
	for _, v := range sMap {
		if v != 0 {
			return false
		}
	}
	return true
}

File: neet/valid_anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        count = collections.defaultdict(int)
        for x in s: count[x] += 1
        for x in t: count[x] -= 1
        return not any(count.values())
        
        # if len(s) != len(t):
        #     return False

        # countS, countT = {}, {}

        # for i in range(len(s)):
        #     countS[s[i]] = 1 + countS.get(s[i], 0)
        #     countT[t[i]] = 1 + countT.get(t[i], 0)
        # return countS == countT

        # return sorted(t) == sorted(s)